sound/soc/sof/ipc4-topology.c

```diff
@@ -1398,12 +1399,24 @@ static int snd_sof_get_nhlt_endpoint_data(struct snd_sof_dev *sdev, struct snd_s
 					   dir, dev_type);
 
 	if (!cfg) {
+		/* no 32-bit DMIC blob found, look for one based on params */
+		if (linktype == SOF_DAI_INTEL_DMIC && params_width(params) != 32) {
+			bit_depth = params_width(params);
+			cfg = intel_nhlt_get_endpoint_blob(sdev->dev, ipc4_data->nhlt,
+							   dai_index, nhlt_type,
+							   bit_depth, bit_depth,
+							   channel_count, sample_rate,
+							   dir, dev_type);
+			if (cfg)
+				goto out;
+		}
```

Comment:
I am still lost with all the formats.

I just looked and back in November we had an issue with two OEMS, one using a 24/32 config and the other a 32/32 config.

What makes us so sure we want to start with the 32-bit case?

Reply:
My point was to check if BOTH the 24/32 and 32/32 configs would be handled if userspace asks for 16-bit capture.

Reply:
for the DMIC, we ignore valid bit depth anyway. So all we care about is the bit depth of 32-bit. So, yes both configs will be handled.

Reply:
ok, thanks @ranj063